// apollo-server-redis/dist/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const Redis = require("redis");
const util_1 = require("util");
class RedisCache {
    constructor(options) {
        this.defaultSetOptions = {
            ttl: 300,
        };
        this.client = Redis.createClient(options);
        this.client.get = util_1.promisify(this.client.get).bind(this.client);
        this.client.set = util_1.promisify(this.client.set).bind(this.client);
        this.client.flushdb = util_1.promisify(this.client.flushdb).bind(this.client);
        this.client.quit = util_1.promisify(this.client.quit).bind(this.client);
    }
    set(key, data, options) {
        return __awaiter(this, void 0, void 0, function* () {
            const { ttl } = Object.assign({}, this.defaultSetOptions, options);
            yield this.client.set(key, data, 'EX', ttl);
        });
    }
    get(key) {
        return __awaiter(this, void 0, void 0, function* () {
            const reply = yield this.client.get(key);
            if (reply !== null) {
                return reply;
            }
            return;
        });
    }
    flush() {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.client.flushdb();
        });
    }
    close() {
        return __awaiter(this, void 0, void 0, function* () {
            yield this.client.quit();
            return;
        });
    }
}
exports.RedisCache = RedisCache;
